hms-app/src/actions/IdeaActions.tsx

```diff
@@ -10,9 +10,7 @@ export const getIdeaList = (hackathonID: string) => {
       'Content-Type': 'application/json',
     },
   })
-    .then((data) => {
-      return data.json()
-    })
+    .then((data) => data.json())
```

Comment:
so in general this is the issue and the reason why you need the `new Date()` everywhere.  The JSON specs do not have a way to represent a Date object, all they have is a string representation of the date.  What you have to do is create the Date objects here.  If you don't then you end up violating the contracts you defined elsewhere when you created the different types.

it might seem tedious to do it, but if you do it here and properly create your objects you only need to do it once and there won't be any weirdness elsewhere

Reply:
so instead of just returning `data.json()` you should specify a strict return type in the function declaration and then do something like (it will be different for each function):
```suggestion
    .then((data) => {
      cosnt json = data.json());
      return {
        id: json.id,
        startDate: new Date(json.startDate);
        ...
      }
    }
```